woocommerce/woocommerce
# Update Jetpack Tracks Client library reference to not have a hardcoded directory path.

This PR updates the way the path to the Jetpack client library is forged so to take in account Jetpack installations that do not live in `plugins/jetpack`. It's the case for the Jetpack Beta Tester plugin, that it will use the `plugins/jetpack-dev` directory when running a development branch of Jetpack.

Note the reference to both `jetpack-dev` and `jetpack` in the directory in this error message:
```
Fatal error: Cannot redeclare jetpack_tracks_record_event_raw() (previously declared in wp-content/plugins/jetpack-dev/_inc/lib/tracks/client.php:29) in wp-content/plugins/jetpack/_inc/lib/tracks/client.php on line 31
```

### All Submissions:






### Changes proposed in this Pull Request:

* Updates  `class-wc-tracks-client.php` to rely on Jetpack's function `jetpack_require_lib_dir()` to figure out the base directory for Jetpack and forging the path to the `tracks/client.php` library.

### How to test the changes in this Pull Request:

1. Start with a site with WooCommerce, [Jetpack Beta Tester plugin](https://jetpack.com/download-jetpack-beta/), and `WP_DEBUG`, `WP_DEBUG_LOG` set to true. ([JN link](https://jurassic.ninja/create?jetpack-beta&woocommerce&wp-debug-log)).
2. Tail the log file. `tail -F wp-content/debug.log`. 
3. Run the setup wizard until the Payment Step. 
4. When clicking there, expect to reach the Shipping step without any error logged.

Before this fix, a fatal error was logged and shown in the screen.

![image](https://user-images.githubusercontent.com/746152/56431277-8a3bae80-629f-11e9-8661-00434405e4a2.png)


### Other information:




### Changelog entry

> Fixed compatibility with the Jetpack Beta Tester plugin when loading the Tracks client library.
